### PR TITLE
Implement  Trace/Log/Error framework, able to collect and store into in standalone mode.

### PR DESCRIPTION
**Pull request by xzxiong**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #3706, #3435

## What this PR does / why we need it:

- Make statement/trace/log/error interrelated by statement_id/span_id.
- store in db, make it can be search by sql cmd.
- error framework provide a unified format to display and optimize error info

### CODE REVIEW

File: pkg/frontend/mysql_cmd_executor.go [hunk 1]
```diff
@@ -133,6 +136,29 @@ func (mce *MysqlCmdExecutor) GetRoutineManager() *RoutineManager {
...
+func (mce *MysqlCmdExecutor) RecordStatement(ctx context.Context, ses *Session, proc *process.Process, sql string, beginIns time.Time) context.Context {
+	statementId := util.Fastrand64()
+	sessInfo := proc.SessionInfo
+	trace.ReportStatement(
+		ctx,
+		&trace.StatementInfo{
+			StatementID:          statementId,
+			SessionID:            sessInfo.GetConnectionID(),
+			TransactionID:        ses.GetTxnHandler().GetTxn().GetID(),
+			Account:              "account",
```
Comment (fengttt) [inline pkg/frontend/mysql_cmd_executor.go:148]: make this as sessionInfo.GetAccount(), even if we return a dummy value there.

File: cmd/generate-config/system_vars_def.toml [hunk 2]
```diff
@@ -327,6 +327,36 @@ values = ["true", "false"]
...
+update-mode = "dynamic"
+
+[[parameter]]
+name = "enableTraceDebug"
+scope = ["global"]
+access = ["file"]
+type = "bool"
+domain-type = "set"
+values = ["false", "true"]
+comment = "default is false. With true, system will check all the children span is ended, which belong to the closing span."
```
Comment (fengttt) [inline cmd/generate-config/system_vars_def.toml:357]: What is a span?    I guess I will read it later in the commit.   However, the name of the parameter, enableTraceDebug, and the comment, does not seem to be the same thing.

**Review by fengttt — COMMENTED** [review]
Small things.   Still looking at the rest.

File: pkg/util/trace/schema.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	logInfoTbl       = "log_info"
+	statementInfoTbl = "statement_info"
+	errorInfoTbl     = "error_info"
+)
+
+const (
+	sqlCreateDBConst = `create database if not exists ` + statsDatabase
+
+	sqlCreateSpanInfoTable = `CREATE TABLE IF NOT EXISTS span_info(
+ span_id BIGINT UNSIGNED,
```
Comment (fengttt) [inline pkg/util/trace/schema.go:24]: span_id and log_info id, they are surrogate keys?    I am not a big fan of this thing.    How would people use this?

Is session id, statement id node_id etc, plus timestamp enough?    Most likely we don't need any primary key for log, and not span_id as foreign key as well.    Most likely we just want to scan this thing.

File: pkg/frontend/mysql_cmd_executor.go (hunk 1 above)
Comment (xzxiong) [reply pkg/frontend/mysql_cmd_executor.go:148]: got it.

File: pkg/util/trace/schema.go (hunk 3 above)
Comment (xzxiong) [reply pkg/util/trace/schema.go:24]: log_info id is useless, will be removed.

Comment (xzxiong) [reply pkg/util/trace/schema.go:24]: in table log_info, span_id is NOT necessary, but can be related to a specific span.
in table span_info, span_id and parent_span_id are `necessary` to build like a call-stack tree.

**Comment by xzxiong** [comment]
@lni @w-zr 麻烦看看 logutil 下的修改, 主要有 3 点
1. `report.go` 提供Encoder和Syncer 支持对 log 进行上报，使得原来的 `logutil.Info/Infof/...` 也可以直接上报日志
2. `dragonboat.go` 提供对 `dragonboat/v4/logger.go` 接口支持
3. `logutil2/api.go` 提供 `context.Context` 的作为第一个参数的 api

**Comment by lni** [comment]
have't got time to read the whole PR, but really love the fact that there is already an adaptor for dragonboat's logging. Thanks for caring for such details.

**Review by fengttt — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

**Review by aptend — APPROVED** [review]

**Comment by xzxiong** [comment]
@mergify refresh

**Review by yingfeng — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Comment by xzxiong** [comment]
@mergify refresh

**Pull request merged**